pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -80,10 +80,33 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		env.SmartRouter.Omit = true
 	}
 
+	if cr.Spec.Environment == api.RhdmProductionImmutable || (isImmutable(cr) && cr.Status.Applied.Objects.Console == nil) {
+		env.Console.Omit = true
+	}
+
 	mergedEnv, err := merge(common, env)
 	if err != nil {
 		return api.Environment{}, err
 	}
+
+	// if bc monitoring is not set, there's no need to set the environment variables below
+	if mergedEnv.Console.Omit {
+		// cleaning console object
+		mergedEnv.Console = api.CustomObject{Omit: true}
+		for _, server := range env.Servers {
+			for _, dc := range server.DeploymentConfigs {
+				newSlice := []corev1.EnvVar{}
+				for _, envvar := range dc.Spec.Template.Spec.Containers[0].Env {
+					_, found := shared.Find(constants.Envs2removeImmutableWithoutConsole, envvar.Name)
+					if !found {
+						newSlice = append(newSlice, envvar)
+					}
+				}
+				dc.Spec.Template.Spec.Containers[0].Env = newSlice
+			}
+		}
+	}
+
```

Comment:
This is something that might make more sense to handle in the go template configs. Instead of removing them after the fact, just never set them if the Console Object doesn't exist.

Something similar to this?
https://github.com/kiegroup/kie-cloud-operator/blob/e16f518f65bccd1f01ecf955fdf30b1ce87a4dc6/config/7.9.0/common.yaml#L200-L203

Reply:
maybe can do this?
#[[if not .Console.Omit]]

Reply:
I avoid to touch the templates because you said that you was planning to remove some logic from templates.
I can move it to the template config if you think better.

Reply:
applied.